src/block/request.rs

```diff
@@ -204,11 +256,142 @@ impl Request {
     }
 }
 
+/// Wraps a block device file for request execution.
+///
+/// # Example
+///
+/// ```rust
+/// # use std::fs::File;
+/// # use vm_memory::GuestAddress;
+/// # use vm_virtio::request::RequestExecutor;
+///
+/// let file = File::create("foo.txt").unwrap();
+/// let request_exec = RequestExecutor::new(file);
+/// ```
+pub struct RequestExecutor<D: DiskFile> {
```

Comment:
Does it makes sense to add the executor to its own module file? I think it's cleaner to look at that way.